tests/unit/swagger/definitionsGeneration/metadata.spec.ts

```diff
@@ -758,6 +758,19 @@ describe('Metadata generation', () => {
       const deprecatedParam2 = method.parameters[2];
       expect(deprecatedParam2.deprecated).to.be.true;
     });
+
+    it('should generate example of 0 for parameter ', () => {
```

Comment:
Adding a metadata spec kinda works around what you're trying to show here. This also passes without any code changes. The bug was the metadata -> openapi transformation, not the user code ->  metadata transformation which occurs before. Have a look at the `schemaDetails` and `schemaDetails3` test infra instead.